sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -30,22 +30,38 @@
      * Logger tag.
      */
     public static final String TAG = AndroidSentryClientFactory.class.getName();
+
     /**
      * Default Buffer directory name.
      */
     private static final String DEFAULT_BUFFER_DIR = "sentry-buffered-events";
 
     private Context ctx;
 
+    /**
+     * Construct an AndroidSentryClientFactory using the base Context from the specified Android Application.
+     *
+     * @param app Android Application
+     */
+    public AndroidSentryClientFactory(Application app) {
+        Log.d(TAG, "Construction of Android Sentry from Android Application.");
+
+        this.ctx = app.getBaseContext();
+    }
+
     /**
      * Construct an AndroidSentryClientFactory using the specified Android Context.
      *
      * @param ctx Android Context.
      */
     public AndroidSentryClientFactory(Context ctx) {
-        Log.d(TAG, "Construction of Android Sentry.");
+        Log.d(TAG, "Construction of Android Sentry from Android Context.");
+
+        this.ctx = ctx;
```

Comment:
This change is a bit risky, as it is going to produce a memory leak if an application is creating a `AndroidSentryClientFactory` with an `Activity` as `ctx`.

Sentry would hold a reference to such `Activity` and any resource tied to it won't be eligible for garbage collection (e.g. the whole view hierarchy, which can be massive in terms of memory occupation).

I am not saying that the previous method of getting hold of a Context was wrong, I just suggested a way for a developer to feed consciously a Context to be used by Sentry.

Reply:
The factories are not long lived and in fact they should not live any longer than it is required to create a new `SentryClient`. I can also wrap `ctx` in a `WeakReference` however.